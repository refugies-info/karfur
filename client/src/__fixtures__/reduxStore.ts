import { RootState } from "services/rootReducer";
import { SearchResultsState } from "services/SearchResults/searchResults.reducer";
import { UserFavoritesState } from "services/UserFavoritesInLocale/UserFavoritesInLocale.reducer";
import { activeThemesMock } from "./activeThemes";

const initialMockUserState = {
  user: null,
  admin: false,
  traducteur: false,
  expertTrad: false,
  contributeur: false,
  membreStruct: false,
  userId: "",
  userFetched: false,
  rolesInStructure: [],
};
const initialMockLangueState = {
  langues: [],
  languei18nCode: "fr",
  showLanguageModal: false,
  showLangModal: false,
};

const initialMockTranslationState = {
  translation: {
    initialText: {},
    translatedText: {},
  },
  translations: [],
};

const initialUserFavoritesState: UserFavoritesState = {
  favorites: [],
  showFavoriteModal: false
}

const initialMockSearchReults: SearchResultsState = {
  results: {
    dispositifs: [],
    demarches: [],
    dispositifsSecondaryTheme: []
  },
  query: {
    search: "",
    departments: [],
    themes: [],
    needs: [],
    age: [],
    frenchLevel: [],
    language: [],
    sort: "date",
    type: "all",
  }
}

export const initialMockStore = {
  user: initialMockUserState,
  langue: initialMockLangueState,
  activeDispositifs: [],
  tts: { ttsActive: false, showAudioSpinner: false },
  userStructure: null,
  selectedDispositif: null,
  loadingStatus: {},
  translation: initialMockTranslationState,
  activeStructures: [],
  selectedStructure: null,
  allDispositifs: [],
  allStructures: [],
  users: [],
  userFavorites: initialUserFavoritesState,
  userContributions: [],
  needs: [],
  themes: {
    activeThemes: activeThemesMock,
    inactiveThemes: []
  },
  searchResults: initialMockSearchReults
};
