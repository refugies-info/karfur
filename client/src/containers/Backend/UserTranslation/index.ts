import UserTranslation from "./UserTranslation";

export default UserTranslation;
