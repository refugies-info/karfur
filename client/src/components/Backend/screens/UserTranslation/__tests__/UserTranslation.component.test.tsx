//@ts-nocheck
import "jest-styled-components";
import { useParams } from "react-router-dom";
import { fetchDispositifsWithTranslationsStatusActionCreator } from "services/DispositifsWithTranslationsStatus/dispositifsWithTranslationsStatus.actions";
import API from "utils/API";
import { dispositifsWithTranslations } from "__fixtures__/dispositifsWithTrad";
import { initialMockStore } from "__fixtures__/reduxStore";
import { wrapWithProvidersAndRenderForTesting } from "../../../../../../jest/lib/wrapWithProvidersAndRender";
import UserTranslation from "../UserTranslation";
jest.mock("next/router", () => require("next-router-mock"));

// Mock history
const replace = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: jest.fn(),
  useHistory: () => ({ replace }),
}));

jest.mock("services/DispositifsWithTranslationsStatus/dispositifsWithTranslationsStatus.actions", () => {
  const actions = jest.requireActual(
    "services/DispositifsWithTranslationsStatus/dispositifsWithTranslationsStatus.actions",
  );
  return {
    fetchDispositifsWithTranslationsStatusActionCreator: jest.fn(
      actions.fetchDispositifsWithTranslationsStatusActionCreator,
    ),
  };
});
jest.mock("utils/API");

describe("user translation", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should return loader langue in url = first selected langue", () => {
    useParams.mockReturnValueOnce({ id: "en" });
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserTranslation,
      reduxState: {
        ...initialMockStore,
        loadingStatus: {
          FETCH_DISPOSITIFS_TRANSLATIONS_STATUS: { isLoading: false },
        },
        user: {
          user: { selectedLanguages: ["en"], _id: "userId" },
        },
      },
    });
    expect(fetchDispositifsWithTranslationsStatusActionCreator).toHaveBeenCalledWith("en");
    expect(API.get_progression).toHaveBeenCalledWith({ onlyTotal: true });
    expect(component).toMatchSnapshot();
  });

  it("should redirect if user has language but not langue in url + snap view without content", () => {
    useParams.mockReturnValueOnce({});
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserTranslation,
      reduxState: {
        ...initialMockStore,
        user: { user: { selectedLanguages: ["en"] } },
        dispositifsWithTranslations: [],
      },
    });
    expect(replace).toHaveBeenCalledWith("/fr/backend/user-translation/en");
    expect(component).toMatchSnapshot();
  });

  it("should redirect if user has no language but langue in url", () => {
    useParams.mockReturnValueOnce({ id: "en" });
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserTranslation,
      reduxState: {
        ...initialMockStore,
        user: { user: { selectedLanguages: [] } },
        dispositifsWithTranslations: [],
      },
    });
    expect(fetchDispositifsWithTranslationsStatusActionCreator).not.toHaveBeenCalled();
    expect(API.get_progression).not.toHaveBeenCalled();
    expect(replace).toHaveBeenCalledWith("/fr/backend/user-translation");
  });

  it("should redirect if user has no language but langue in url + snap no langue", () => {
    useParams.mockReturnValueOnce({ id: "test" });
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserTranslation,
      reduxState: {
        ...initialMockStore,
        user: { user: { selectedLanguages: [] } },
        dispositifsWithTranslations: [{ _id: "id" }],
      },
    });
    expect(fetchDispositifsWithTranslationsStatusActionCreator).not.toHaveBeenCalled();
    expect(API.get_progression).not.toHaveBeenCalled();
    expect(replace).toHaveBeenCalledWith("/fr/backend/user-translation");
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with non expert, non admin", () => {
    useParams.mockReturnValueOnce({ id: "en" });
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserTranslation,
      reduxState: {
        ...initialMockStore,
        user: {
          user: { selectedLanguages: [{ i18nCode: "en", _id: "idEn" }] },
          expertTrad: false,
        },
        dispositifsWithTranslations,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with expert, non admin", () => {
    useParams.mockReturnValueOnce({ id: "en" });
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserTranslation,
      reduxState: {
        ...initialMockStore,
        user: {
          user: { selectedLanguages: [{ i18nCode: "en", _id: "idEn" }] },
          expertTrad: true,
        },
        dispositifsWithTranslations,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with expert, admin", () => {
    useParams.mockReturnValueOnce({ id: "en" });
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserTranslation,
      reduxState: {
        ...initialMockStore,
        user: {
          user: { selectedLanguages: [{ i18nCode: "en" }] },
          expertTrad: true,
          admin: true,
        },
        dispositifsWithTranslations,
      },
      compProps: {},
    });

    expect(component).toMatchSnapshot();
  });
});
