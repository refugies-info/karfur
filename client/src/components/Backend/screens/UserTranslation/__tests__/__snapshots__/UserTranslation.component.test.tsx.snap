// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user translation should redirect if user has language but not langue in url + snap view without content 1`] = `
.c1 {
  font-weight: bold;
  font-size: 27.7097px;
  line-height: 35px;
  margin-bottom: 40px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  background: #ffffff;
  border-radius: 12px;
  width: 1040px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  font-size: 32px;
  line-height: 40px;
  color: #5e5e5e;
  margin-top: 60px;
  margin-bottom: 20px;
}

.c4 {
  font-size: 18px;
  line-height: 23px;
  color: #5e5e5e;
  margin-bottom: 40px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 60px;
}

<div
  className="main"
>
  <div
    className="container"
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        Vous n'avez pas traduit de fiches pour le moment.
      </div>
      <div
        className="c2"
      >
        <div
          className="c3"
        >
          Vous parlez une autre langue ?
        </div>
        <div
          className="c4"
        >
          Aidez à traduire les contenus à votre rythme.
        </div>
        <div
          className="c5"
        >
          <button
            className="btn dark me-2"
            onClick={[Function]}
          >
            <img
              alt="a"
              data-nimg="1"
              decoding="async"
              height={21}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={21}
            />
            Commencer à traduire
          </button>
          <button
            className="btn tuto ms-2"
            onClick={[Function]}
          >
            <span
              className="me-2 icon"
            >
              <i
                dangerouslySetInnerHTML={
                  {
                    "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-video-outline" fill="#fff"><g data-name="Layer 2"><g data-name="video"><rect width="24" height="24" opacity="0"/><path d="M21 7.15a1.7 1.7 0 0 0-1.85.3l-2.15 2V8a3 3 0 0 0-3-3H5a3 3 0 0 0-3 3v8a3 3 0 0 0 3 3h9a3 3 0 0 0 3-3v-1.45l2.16 2a1.74 1.74 0 0 0 1.16.45 1.68 1.68 0 0 0 .69-.15 1.6 1.6 0 0 0 1-1.48V8.63A1.6 1.6 0 0 0 21 7.15zM15 16a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V8a1 1 0 0 1 1-1h9a1 1 0 0 1 1 1zm5-1.4L17.19 12 20 9.4z"/></g></g></svg>",
                  }
                }
                style={
                  {
                    "height": "20px",
                    "lineHeight": "20px",
                  }
                }
              />
            </span>
            Explications
          </button>
        </div>
      </div>
    </div>
    <button
      aria-controls="user-pseudo-modal"
      aria-hidden={true}
      className="fr-btn fr-hidden"
      data-fr-opened={false}
      id="user-pseudo-modal-hidden-control-button"
      style={{}}
      tabIndex={-1}
      type="button"
    >
       
    </button>
    <dialog
      aria-labelledby="fr-modal-title-user-pseudo-modal"
      className="fr-modal"
      data-fr-concealing-backdrop={true}
      id="user-pseudo-modal"
      role="dialog"
    >
      <div
        className="fr-container fr-container--fluid fr-container-md"
      >
        <div
          className="fr-grid-row fr-grid-row--center"
        >
          <div
            className="fr-col-12 fr-col-md-8 fr-col-lg-6"
          >
            <div
              className="fr-modal__body"
            >
              <div
                className="fr-modal__header"
              >
                <button
                  aria-controls="user-pseudo-modal"
                  className="fr-btn--close fr-btn"
                  title="Fermer"
                  type="button"
                >
                  Fermer
                </button>
              </div>
              <div
                className="fr-modal__content"
              >
                <h1
                  className="fr-modal__title"
                  id="fr-modal-title-user-pseudo-modal"
                >
                  Choisissez un pseudonyme
                </h1>
                <p>
                  Au pied de chaque fiche, nous affichons les contributeurs ayant participé à sa rédaction et sa traduction.
                </p>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="fr-input-group mt-12 mb-0"
                  >
                    <label
                      className="fr-label"
                      htmlFor="input-:r4:"
                    >
                      Votre pseudonyme
                      <span
                        className="fr-hint-text"
                      >
                        N'indiquez pas de coordonnées personnelles comme votre adresse email. Exemples : Guillaume-afpa, cidff13, sarah-trad, Nora78.
                      </span>
                    </label>
                    <input
                      aria-describedby="input-:r4:-desc-error"
                      autoFocus={true}
                      className="fr-input"
                      id="input-:r4:"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                  <button
                    className="fr-btn fr-icon-arrow-right-line fr-btn--icon-right button mt-12"
                    disabled={false}
                    id="fr-button-:r5:"
                    style={{}}
                    type="submit"
                  >
                    Suivant
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</div>
`;

exports[`user translation should redirect if user has no language but langue in url + snap no langue 1`] = `
.c1 {
  font-weight: bold;
  font-size: 27.7097px;
  line-height: 35px;
  margin-bottom: 40px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  background: #ffffff;
  border-radius: 12px;
  width: 1040px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  font-size: 32px;
  line-height: 40px;
  color: #5e5e5e;
  margin-top: 60px;
  margin-bottom: 20px;
}

.c4 {
  font-size: 18px;
  line-height: 23px;
  color: #5e5e5e;
  margin-bottom: 40px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 60px;
}

<div
  className="main"
>
  <div
    className="container"
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        Vous n'avez pas traduit de fiches pour le moment.
      </div>
      <div
        className="c2"
      >
        <div
          className="c3"
        >
          Vous parlez une autre langue ?
        </div>
        <div
          className="c4"
        >
          Aidez à traduire les contenus à votre rythme.
        </div>
        <div
          className="c5"
        >
          <button
            className="btn dark me-2"
            onClick={[Function]}
          >
            <img
              alt="a"
              data-nimg="1"
              decoding="async"
              height={21}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={21}
            />
            Commencer à traduire
          </button>
          <button
            className="btn tuto ms-2"
            onClick={[Function]}
          >
            <span
              className="me-2 icon"
            >
              <i
                dangerouslySetInnerHTML={
                  {
                    "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-video-outline" fill="#fff"><g data-name="Layer 2"><g data-name="video"><rect width="24" height="24" opacity="0"/><path d="M21 7.15a1.7 1.7 0 0 0-1.85.3l-2.15 2V8a3 3 0 0 0-3-3H5a3 3 0 0 0-3 3v8a3 3 0 0 0 3 3h9a3 3 0 0 0 3-3v-1.45l2.16 2a1.74 1.74 0 0 0 1.16.45 1.68 1.68 0 0 0 .69-.15 1.6 1.6 0 0 0 1-1.48V8.63A1.6 1.6 0 0 0 21 7.15zM15 16a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V8a1 1 0 0 1 1-1h9a1 1 0 0 1 1 1zm5-1.4L17.19 12 20 9.4z"/></g></g></svg>",
                  }
                }
                style={
                  {
                    "height": "20px",
                    "lineHeight": "20px",
                  }
                }
              />
            </span>
            Explications
          </button>
        </div>
      </div>
    </div>
    <button
      aria-controls="user-pseudo-modal"
      aria-hidden={true}
      className="fr-btn fr-hidden"
      data-fr-opened={false}
      id="user-pseudo-modal-hidden-control-button"
      style={{}}
      tabIndex={-1}
      type="button"
    >
       
    </button>
    <dialog
      aria-labelledby="fr-modal-title-user-pseudo-modal"
      className="fr-modal"
      data-fr-concealing-backdrop={true}
      id="user-pseudo-modal"
      role="dialog"
    >
      <div
        className="fr-container fr-container--fluid fr-container-md"
      >
        <div
          className="fr-grid-row fr-grid-row--center"
        >
          <div
            className="fr-col-12 fr-col-md-8 fr-col-lg-6"
          >
            <div
              className="fr-modal__body"
            >
              <div
                className="fr-modal__header"
              >
                <button
                  aria-controls="user-pseudo-modal"
                  className="fr-btn--close fr-btn"
                  title="Fermer"
                  type="button"
                >
                  Fermer
                </button>
              </div>
              <div
                className="fr-modal__content"
              >
                <h1
                  className="fr-modal__title"
                  id="fr-modal-title-user-pseudo-modal"
                >
                  Choisissez un pseudonyme
                </h1>
                <p>
                  Au pied de chaque fiche, nous affichons les contributeurs ayant participé à sa rédaction et sa traduction.
                </p>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="fr-input-group mt-12 mb-0"
                  >
                    <label
                      className="fr-label"
                      htmlFor="input-:ra:"
                    >
                      Votre pseudonyme
                      <span
                        className="fr-hint-text"
                      >
                        N'indiquez pas de coordonnées personnelles comme votre adresse email. Exemples : Guillaume-afpa, cidff13, sarah-trad, Nora78.
                      </span>
                    </label>
                    <input
                      aria-describedby="input-:ra:-desc-error"
                      autoFocus={true}
                      className="fr-input"
                      id="input-:ra:"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                  <button
                    className="fr-btn fr-icon-arrow-right-line fr-btn--icon-right button mt-12"
                    disabled={false}
                    id="fr-button-:rb:"
                    style={{}}
                    type="submit"
                  >
                    Suivant
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</div>
`;

exports[`user translation should render correctly with expert, admin 1`] = `
<div
  className="main"
>
  <div
    className="container"
  >
    <button
      aria-controls="user-pseudo-modal"
      aria-hidden={true}
      className="fr-btn fr-hidden"
      data-fr-opened={false}
      id="user-pseudo-modal-hidden-control-button"
      style={{}}
      tabIndex={-1}
      type="button"
    >
       
    </button>
    <dialog
      aria-labelledby="fr-modal-title-user-pseudo-modal"
      className="fr-modal"
      data-fr-concealing-backdrop={true}
      id="user-pseudo-modal"
      role="dialog"
    >
      <div
        className="fr-container fr-container--fluid fr-container-md"
      >
        <div
          className="fr-grid-row fr-grid-row--center"
        >
          <div
            className="fr-col-12 fr-col-md-8 fr-col-lg-6"
          >
            <div
              className="fr-modal__body"
            >
              <div
                className="fr-modal__header"
              >
                <button
                  aria-controls="user-pseudo-modal"
                  className="fr-btn--close fr-btn"
                  title="Fermer"
                  type="button"
                >
                  Fermer
                </button>
              </div>
              <div
                className="fr-modal__content"
              >
                <h1
                  className="fr-modal__title"
                  id="fr-modal-title-user-pseudo-modal"
                >
                  Choisissez un pseudonyme
                </h1>
                <p>
                  Au pied de chaque fiche, nous affichons les contributeurs ayant participé à sa rédaction et sa traduction.
                </p>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="fr-input-group mt-12 mb-0"
                  >
                    <label
                      className="fr-label"
                      htmlFor="input-:rj:"
                    >
                      Votre pseudonyme
                      <span
                        className="fr-hint-text"
                      >
                        N'indiquez pas de coordonnées personnelles comme votre adresse email. Exemples : Guillaume-afpa, cidff13, sarah-trad, Nora78.
                      </span>
                    </label>
                    <input
                      aria-describedby="input-:rj:-desc-error"
                      autoFocus={true}
                      className="fr-input"
                      id="input-:rj:"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                  <button
                    className="fr-btn fr-icon-arrow-right-line fr-btn--icon-right button mt-12"
                    disabled={false}
                    id="fr-button-:rk:"
                    style={{}}
                    type="submit"
                  >
                    Suivant
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</div>
`;

exports[`user translation should render correctly with expert, non admin 1`] = `
<div
  className="main"
>
  <div
    className="container"
  >
    <button
      aria-controls="user-pseudo-modal"
      aria-hidden={true}
      className="fr-btn fr-hidden"
      data-fr-opened={false}
      id="user-pseudo-modal-hidden-control-button"
      style={{}}
      tabIndex={-1}
      type="button"
    >
       
    </button>
    <dialog
      aria-labelledby="fr-modal-title-user-pseudo-modal"
      className="fr-modal"
      data-fr-concealing-backdrop={true}
      id="user-pseudo-modal"
      role="dialog"
    >
      <div
        className="fr-container fr-container--fluid fr-container-md"
      >
        <div
          className="fr-grid-row fr-grid-row--center"
        >
          <div
            className="fr-col-12 fr-col-md-8 fr-col-lg-6"
          >
            <div
              className="fr-modal__body"
            >
              <div
                className="fr-modal__header"
              >
                <button
                  aria-controls="user-pseudo-modal"
                  className="fr-btn--close fr-btn"
                  title="Fermer"
                  type="button"
                >
                  Fermer
                </button>
              </div>
              <div
                className="fr-modal__content"
              >
                <h1
                  className="fr-modal__title"
                  id="fr-modal-title-user-pseudo-modal"
                >
                  Choisissez un pseudonyme
                </h1>
                <p>
                  Au pied de chaque fiche, nous affichons les contributeurs ayant participé à sa rédaction et sa traduction.
                </p>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="fr-input-group mt-12 mb-0"
                  >
                    <label
                      className="fr-label"
                      htmlFor="input-:rg:"
                    >
                      Votre pseudonyme
                      <span
                        className="fr-hint-text"
                      >
                        N'indiquez pas de coordonnées personnelles comme votre adresse email. Exemples : Guillaume-afpa, cidff13, sarah-trad, Nora78.
                      </span>
                    </label>
                    <input
                      aria-describedby="input-:rg:-desc-error"
                      autoFocus={true}
                      className="fr-input"
                      id="input-:rg:"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                  <button
                    className="fr-btn fr-icon-arrow-right-line fr-btn--icon-right button mt-12"
                    disabled={false}
                    id="fr-button-:rh:"
                    style={{}}
                    type="submit"
                  >
                    Suivant
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</div>
`;

exports[`user translation should render correctly with non expert, non admin 1`] = `
<div
  className="main"
>
  <div
    className="container"
  >
    <button
      aria-controls="user-pseudo-modal"
      aria-hidden={true}
      className="fr-btn fr-hidden"
      data-fr-opened={false}
      id="user-pseudo-modal-hidden-control-button"
      style={{}}
      tabIndex={-1}
      type="button"
    >
       
    </button>
    <dialog
      aria-labelledby="fr-modal-title-user-pseudo-modal"
      className="fr-modal"
      data-fr-concealing-backdrop={true}
      id="user-pseudo-modal"
      role="dialog"
    >
      <div
        className="fr-container fr-container--fluid fr-container-md"
      >
        <div
          className="fr-grid-row fr-grid-row--center"
        >
          <div
            className="fr-col-12 fr-col-md-8 fr-col-lg-6"
          >
            <div
              className="fr-modal__body"
            >
              <div
                className="fr-modal__header"
              >
                <button
                  aria-controls="user-pseudo-modal"
                  className="fr-btn--close fr-btn"
                  title="Fermer"
                  type="button"
                >
                  Fermer
                </button>
              </div>
              <div
                className="fr-modal__content"
              >
                <h1
                  className="fr-modal__title"
                  id="fr-modal-title-user-pseudo-modal"
                >
                  Choisissez un pseudonyme
                </h1>
                <p>
                  Au pied de chaque fiche, nous affichons les contributeurs ayant participé à sa rédaction et sa traduction.
                </p>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="fr-input-group mt-12 mb-0"
                  >
                    <label
                      className="fr-label"
                      htmlFor="input-:rd:"
                    >
                      Votre pseudonyme
                      <span
                        className="fr-hint-text"
                      >
                        N'indiquez pas de coordonnées personnelles comme votre adresse email. Exemples : Guillaume-afpa, cidff13, sarah-trad, Nora78.
                      </span>
                    </label>
                    <input
                      aria-describedby="input-:rd:-desc-error"
                      autoFocus={true}
                      className="fr-input"
                      id="input-:rd:"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                  <button
                    className="fr-btn fr-icon-arrow-right-line fr-btn--icon-right button mt-12"
                    disabled={false}
                    id="fr-button-:re:"
                    style={{}}
                    type="submit"
                  >
                    Suivant
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</div>
`;

exports[`user translation should return loader langue in url = first selected langue 1`] = `
.c1 {
  font-weight: bold;
  font-size: 27.7097px;
  line-height: 35px;
  margin-bottom: 40px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  background: #ffffff;
  border-radius: 12px;
  width: 1040px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  font-size: 32px;
  line-height: 40px;
  color: #5e5e5e;
  margin-top: 60px;
  margin-bottom: 20px;
}

.c4 {
  font-size: 18px;
  line-height: 23px;
  color: #5e5e5e;
  margin-bottom: 40px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 60px;
}

<div
  className="main"
>
  <div
    className="container"
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        Vous n'avez pas traduit de fiches pour le moment.
      </div>
      <div
        className="c2"
      >
        <div
          className="c3"
        >
          Vous parlez une autre langue ?
        </div>
        <div
          className="c4"
        >
          Aidez à traduire les contenus à votre rythme.
        </div>
        <div
          className="c5"
        >
          <button
            className="btn dark me-2"
            onClick={[Function]}
          >
            <img
              alt="a"
              data-nimg="1"
              decoding="async"
              height={21}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={21}
            />
            Commencer à traduire
          </button>
          <button
            className="btn tuto ms-2"
            onClick={[Function]}
          >
            <span
              className="me-2 icon"
            >
              <i
                dangerouslySetInnerHTML={
                  {
                    "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-video-outline" fill="#fff"><g data-name="Layer 2"><g data-name="video"><rect width="24" height="24" opacity="0"/><path d="M21 7.15a1.7 1.7 0 0 0-1.85.3l-2.15 2V8a3 3 0 0 0-3-3H5a3 3 0 0 0-3 3v8a3 3 0 0 0 3 3h9a3 3 0 0 0 3-3v-1.45l2.16 2a1.74 1.74 0 0 0 1.16.45 1.68 1.68 0 0 0 .69-.15 1.6 1.6 0 0 0 1-1.48V8.63A1.6 1.6 0 0 0 21 7.15zM15 16a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V8a1 1 0 0 1 1-1h9a1 1 0 0 1 1 1zm5-1.4L17.19 12 20 9.4z"/></g></g></svg>",
                  }
                }
                style={
                  {
                    "height": "20px",
                    "lineHeight": "20px",
                  }
                }
              />
            </span>
            Explications
          </button>
        </div>
      </div>
    </div>
    <button
      aria-controls="user-pseudo-modal"
      aria-hidden={true}
      className="fr-btn fr-hidden"
      data-fr-opened={false}
      id="user-pseudo-modal-hidden-control-button"
      style={{}}
      tabIndex={-1}
      type="button"
    >
       
    </button>
    <dialog
      aria-labelledby="fr-modal-title-user-pseudo-modal"
      className="fr-modal"
      data-fr-concealing-backdrop={true}
      id="user-pseudo-modal"
      role="dialog"
    >
      <div
        className="fr-container fr-container--fluid fr-container-md"
      >
        <div
          className="fr-grid-row fr-grid-row--center"
        >
          <div
            className="fr-col-12 fr-col-md-8 fr-col-lg-6"
          >
            <div
              className="fr-modal__body"
            >
              <div
                className="fr-modal__header"
              >
                <button
                  aria-controls="user-pseudo-modal"
                  className="fr-btn--close fr-btn"
                  title="Fermer"
                  type="button"
                >
                  Fermer
                </button>
              </div>
              <div
                className="fr-modal__content"
              >
                <h1
                  className="fr-modal__title"
                  id="fr-modal-title-user-pseudo-modal"
                >
                  Choisissez un pseudonyme
                </h1>
                <p>
                  Au pied de chaque fiche, nous affichons les contributeurs ayant participé à sa rédaction et sa traduction.
                </p>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="fr-input-group mt-12 mb-0"
                  >
                    <label
                      className="fr-label"
                      htmlFor="input-:r1:"
                    >
                      Votre pseudonyme
                      <span
                        className="fr-hint-text"
                      >
                        N'indiquez pas de coordonnées personnelles comme votre adresse email. Exemples : Guillaume-afpa, cidff13, sarah-trad, Nora78.
                      </span>
                    </label>
                    <input
                      aria-describedby="input-:r1:-desc-error"
                      autoFocus={true}
                      className="fr-input"
                      id="input-:r1:"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                  <button
                    className="fr-btn fr-icon-arrow-right-line fr-btn--icon-right button mt-12"
                    disabled={false}
                    id="fr-button-:r2:"
                    style={{}}
                    type="submit"
                  >
                    Suivant
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </dialog>
  </div>
</div>
`;
